src/main/java/com/zendesk/maxwell/schema/columndef/ColumnDefWithLength.java

```diff
@@ -0,0 +1,73 @@
+package com.zendesk.maxwell.schema.columndef;
+
+import java.sql.Time;
+import java.sql.Timestamp;
+
+public abstract class ColumnDefWithLength extends ColumnDef {
+	protected Long columnLength;
+
+	protected static ThreadLocal<StringBuilder> threadLocalBuilder = new ThreadLocal<StringBuilder>() {
+		@Override
+		protected StringBuilder initialValue() {
+			return new StringBuilder();
+		}
+
+		@Override
+		public StringBuilder get() {
+			StringBuilder b = super.get();
+			b.setLength(0);
+			return b;
+		}
+	};
+
+	public ColumnDefWithLength(String name, String type, int pos, Long columnLength) {
+		super(name, type, pos);
+		if ( columnLength == null )
+			this.columnLength = 0L;
+		else
+			this.columnLength = columnLength;
+	}
+
+	public Long getColumnLength() { return columnLength ; }
+
+	@Override
+	public String toSQL(Object value) {
+		return "'" + formatValue(value) + "'";
+	}
+
+
+	@Override
+	public Object asJSON(Object value) {
+		return formatValue(value);
+	}
+
+	protected abstract String formatValue(Object value);
+
+	// Truncates the number of nano secs to the column length.
+	// 123 456 789 nano secs for:
+	// - col length 3 -> 123 ;
+	// - col length 6 -> 123 456.
+	// 123 456 nano secs (123 micro secs):
+	// - col length 3 -> 0 (milli secs)
+	// - col length 6 -> 123 (micro secs).
+	protected static int truncateNanosToColumnLength(int nanos, Long columnLength) {
+		int micros = nanos / 1000;
+		// 6 is the max precision of datetime2 in MysQL
+		int divideBy = ((int) Math.pow(10, 6 - columnLength));
+		return micros / divideBy;
+	}
+
+	protected static String objectWithPrecisionToString(String value, Timestamp t, Long columnLength) {
+		int fraction = truncateNanosToColumnLength(t.getNanos(), columnLength);
+		if (fraction == 0)
+			return value;
+
+		String strFormat = "%0" + columnLength + "d";
```

Comment:
You're right, but actually I may have to keep the division and round it to the upper integer. MySQL does that: http://dev.mysql.com/doc/refman/5.6/en/fractional-seconds.html

`.777` for `time(2)` gives `.78`